dask/dataframe/core.py

```diff
@@ -2605,8 +2605,9 @@ def rename(self, index=None, inplace=False, sorted_index=False):
         pandas.Series.rename
         """
         from pandas.api.types import is_scalar, is_list_like, is_dict_like
+        from dask.dataframe import Index
 
-        if is_scalar(index) or (is_list_like(index) and not is_dict_like(index)):
```

Comment:
Should be something like `not isinstance(index, Index)`. You might need this condition in addition to the is_list_like and is_dict_like checks.

Reply:
I see what you mean...adding ```and not isinstance(index, Index)``` to the end of the if statement will avoid the error when passing a dd.Series index to ```rename```.  I was originally trying to make the conditions noted in the docstring explicit in the else statement:  ```If dict-like or callable, the transformation is applied to the index```.